### DISCUSSION TITLE
Organizational structure of source code

### DISCUSSION THREAD

**Discussion by alexschroeter** [arkitektio/arkitektio.github.io#4] [General]
To reduce the barrier of entry for people unfamiliar with Arkitekt, we should structure the repository in a way that reflects the needs of the users.

I think the interested parties will be:
- Users
  - Has Orkestrator installed on their local machine or uses the web view of a hosted instance
  - May add apps to their instance that do what they want
  - In general, doesn't care as long as it works
- Administrators
  - Hosts an Arkitekt instance for themselves or others
  - Doesn't necessarily develop new apps but manages the available apps in the instance
- Developers
  - Writes new or fixes apps
  - Might not even be using Arkitekt but wants others to use their tool in Arkitekt
  - Might also develops core components of Arkitekt

I think this list lends itself to three organizational units:
- Server
- Client
- Apps

Ideally, I envision a structure like this:
- org:Arkitekt
  - repo:Documentation
  - org:Server
    - repo:Mikro
    - repo:Konstruktor
    - ...
  - org:Client
    - repo:Arkitekt CLI
    - repo:Orkestrator
    - ...
  - org:Apps
    - repo:Segmentor
    - repo:Cellpose
    - ...

but since Github Organizations are not stackable, I believe we have the following options:
- Leave the overarching organization and have three Github Organizations (Arkitektio-Server, Arkitektio-Clients, Arkitektio-Apps)
- Keep them all in one repository and make their role clear via the name or add labels

I favor the split because a separation between Apps and Core will make the concept clearer to new users and separate errors in Apps and Arkitekt. We will fix broken Apps, especially in the early stages of adoption. Still, when we hit critical mass, Apps should be developed and maintained by the community, not by core developers.

**Comment by jhnnsrs** [comment]
Hey thanks for setting this up :)
I agree that a split in a core organization and a "community organization" would make a lot of sense, and that apps themselves should be in a dedicated organization. But i'd prefer a single organization for the Arkitekt Core / Server and SDK together, so that one could easily track bugs that would affect both? What do yout hink?

**Reply by alexschroeter** [reply]
So if I understand you correctly, you would want "Arkitektio" to be the main repository which will hold everything but the Apps. This would work for me since my main concern is the splitting of the Apps. 

Do you have a specific example of the "community organization" in mind other than Apps? Otherwise, I would propose creating the "Arkitektio-Apps" organization and moving them to this directory.

**Reply by jhnnsrs** [reply]
I think that sounds great to me :) Would you like to create one?

**Comment by alexschroeter** [comment]
created https://github.com/arkitektio-apps organization.
